### ISSUE TITLE
Changes to YLE API

### ISSUE THREAD

**Issue by FunctionalHacker** [hirsivaja/plugin.video.areena#44]
Hi, I received an email today concerning the changes of the YLE API. Is this something that will affect this plugin?

> Hello, user of Yle API!

We at Yle have been taking a look at open interfaces based on their use. We are disabling some of the lesser-used interfaces and focusing our maintenance resources on the critical functions. For this reason, programme data and stream distribution data interfaces will be removed from the Yle APIs during the spring.

If your current API solution is of social or business importance to you or your company, you can request access to an internal API whose interfaces are different from the current implementation. At this stage, we are accepting professional partners who use APIs to maintain services intended for a broad user base as internal API users.

We are currently mapping out the use of open interfaces, and we hope that you contact us via this form if the use of an internal API is necessary to your operations:
https://forms.gle/hc6yFMyAvydgcq1JA

We are aiming to provide access to the new interface to everyone whose operations would be significantly hindered from not being able to use an interface.

URLs of the APIs being disabled:

https://external.api.yle.fi/v1/programs/items.json

https://external.api.yle.fi/v1/programs/items/{id}.json

https://external.api.yle.fi/v1/series/items.json

https://external.api.yle.fi/v1/series/items/{id}.json

https://external.api.yle.fi/v1/programs/categories.json

https://external.api.yle.fi/v1/programs/services.json

https://external.api.yle.fi/v1/programs/services/{id}.json

https://external.api.yle.fi/v1/programs/schedules.json

https://external.api.yle.fi/v1/programs/schedules/now.json

https://external.api.yle.fi/v1/programs/nowplaying/{service}.json

https://external.api.yle.fi/v1/media/playouts.json

https://external.api.yle.fi/v1/tracking/streamstart

**Comment by hirsivaja** [comment]
Yes, those changes would disable most of this plugins functionality

**Comment by FunctionalHacker** [comment]
So basically everyone using this plugin would need to contact yle to have then enable the endpoints for their user. I will contact Yle about this and see how it goes

**Comment by KasteluKannu** [comment]
> 
> 
> So basically everyone using this plugin would need to contact Yle to have them enable the endpoints for their user. I will contact Yle about this and see how it goes

Could you tell here what they reply? Thanks.

It would be good to have also in the future a working Yle Areena plugin for KODI.

**Comment by T3m3z** [comment]
Indeed. I use this Areena plugin on a daily basis and it would be a shame to lose such a nice functionality. If there is need, I can also participate in efforts keeping this plugin available.

**Comment by hirsivaja** [comment]
I created a poll to see how much users we actually have.
https://github.com/hirsivaja/plugin.video.areena/issues/45

Please go vote.

If you see the external YLE API to be valuable then you send YLE feedback on that.
https://developer.yle.fi/ota-yhteytta/index.html

**Comment by kaskelotti** [comment]
If YLE allows access to the internal API, I'm also available to participate in making the required changes to the plugin code.

**Comment by kaskelotti** [comment]
My add-on stopped working this week, I think on Tuesday. Is it just me or did YLE block their API?

@FunctionalHacker did you get any response from YLE regarding the use of internal APIs?

**Comment by Tea-pot** [comment]
Hey, the same here. Plugin just stop working this week on my Pi3b+ OSMC (Kodi). I just contacted YLE, but they send me an information that the URLs are disabled now.

**Comment by T3m3z** [comment]
Yeah, APIs seem to be inactive now.

One theoretical "workaround" to allow playing Yle Areena videos could be using python tool yle-dl to retrieve address of the video (possible through command line parameters) and play it through Kodi. This of course isn't as handy as one needs to know programid/url in advance and this might not be ideal.

What I am trying to say is that it still might be possible to play Yle Areena videos but with reduced (non-existent) search capabilities inside Kodi UI.

**Comment by hirsivaja** [comment]
It seems that the plugin is not working for anyone anymore (for me it is not working either).

The only part still working is the live-tv. I have made one last commit to the plugin removing all the non-working parts.

I will not continue the plugin development anymore.

**Comment by hirsivaja** [comment]
Hi @T3m3z,
I think it could be possible to do a plugin that parses the information from the Areena web-page. Before we had the API access I remember there was actually a plugin that worked just like that.

Just giving this as an idea, not interested in implementing it myself.

**Comment by j-paulus** [comment]
@hirsivaja  Many thanks for your contribution with the plugin as long as it worked, highly appreciated! It is unfortunate that YLE decided to close the API. At least for our family this means no YLE anymore until some suitable alternative is found.

**Comment by teepean** [comment]
I am just wondering how our TV's Areena plugin still works as it got last update several months ago.

**Comment by Tea-pot** [comment]
Hey, thanks for the update.
Thanks Toni for your work on this. Highly appreciated. That was great being
able to run YLE from Pi3b+ with your plugin. Thank you.

Hopefully not all is lost yet.

**Comment by hirsivaja** [comment]
> 
> 
> I am just wondering how our TV's Areena plugin still works as it got last update several months ago.

The plugin in the TV does not use the external API this plugin used

**Comment by bzoltan1** [comment]
> It seems that the plugin is not working for anyone anymore (for me it is not working either).
> 
> The only part still working is the live-tv. I have made one last commit to the plugin removing all the non-working parts.
> 
> I will not continue the plugin development anymore.

Thank you for your contribution. Me and my family watched Yle news and and children's programs for years with this plugin on RPi devices. It is really sad that Yle closed these APIs.

**Comment by ikke-t** [comment]
Thanks, this plugin has been in active use for years in our family. Really disappointing decision from yle. I wonder why.

**Comment by karsta62** [comment]
> It seems that the plugin is not working for anyone anymore (for me it is not working either).
> 
> The only part still working is the live-tv. I have made one last commit to the plugin removing all the non-working parts.
> 
> I will not continue the plugin development anymore.

In my libreelec setup on rpi2 everything else works but video playback. That probably includes the URL:s since search works as well. So the yle-dl thing would be insteresting, though I'm not able to do it...

**Comment by finnhubb** [comment]
> I think it could be possible to do a plugin that parses the information from the Areena web-page.
> 
> Just giving this as an idea, not interested in implementing it myself.

I have implemented this approach here: [plugin.video.areena](https://github.com/finnhubb/plugin.video.areena)
